subprojects/core/src/integTest/groovy/org/gradle/process/internal/health/memory/MemoryStatusUpdateIntegrationTest.groovy

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.process.internal.health.memory
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.daemon.DaemonLogsAnalyzer
+import org.gradle.integtests.fixtures.daemon.DaemonsFixture
+import org.gradle.integtests.fixtures.executer.GradleContextualExecuter
+import spock.lang.IgnoreIf
+import spock.lang.Timeout
+
+@IgnoreIf({ GradleContextualExecuter.daemon })
+class MemoryStatusUpdateIntegrationTest extends AbstractIntegrationSpec {
+
+    @Timeout(20)
+    def "can register a listener for JVM and OS memory status update events (daemon)"() {
+        given:
+        executer.requireIsolatedDaemons()
+        executer.requireDaemon()
+
+        and:
+        buildFile << waitForMemoryEventsTask()
+
+        when:
+        executer.withTasks("waitForMemoryEvents").run()
+
+        then:
+        daemons.daemon.log.contains jvmLogStatement()
+        daemons.daemon.log.contains osLogStatement()
+    }
+
+    @Timeout(20)
+    def "can register a listener for JVM and OS memory status update events (embedded)"() {
+        given:
+        buildFile << waitForMemoryEventsTask()
+
+        when:
+        succeeds 'waitForMemoryEvents'
+
+        then:
+        result.output.contains jvmLogStatement()
```

Comment:
I'm curious why this is different from the daemon case, i.e. why can't this be one test which uses whatever executor is currently being used?